llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -2535,6 +2535,11 @@ bool RISCVTTIImpl::isLegalMaskedExpandLoad(Type *DataTy, Align Alignment) {
   return true;
 }
 
+bool RISCVTTIImpl::enableConditionalScalarAssignmentVectorization() const {
+  return ST->hasVInstructions() &&
+         ST->getProcFamily() == RISCVSubtarget::SiFive7;
```

Comment:
It would be great to not limit this to a single processor family. IIUC there's nothing target-specific about the feature and the cost model should ideally accurately compute the cost, without needing a separate TTI hook?